src/main/java/com/microsoft/graph/core/IClientConfig.java

```diff
@@ -53,6 +53,15 @@
      */
     IHttpProvider getHttpProvider();
 
+    /**
+     * Gets the HTTP provider
+     * 
+     * @param httpClient the http client to pass to the http provider when building it
+     * @param <T1> the http client type
+     * @return the HTTP provider
+     */
+    <T1> IHttpProvider getHttpProvider(final T1 httpClient);
```

Comment:
How come we aren't seeing a compile time issue with this not being implemented in [DefaultClientConfig.java](https://github.com/microsoftgraph/msgraph-sdk-java/blob/91146e651dfbbaf5e7fa148fd16a1f9688371fa8/src/main/java/com/microsoft/graph/core/DefaultClientConfig.java)

Reply:
because that class is abstract?

Reply:
no, actually because I've implemented it https://github.com/microsoftgraph/msgraph-sdk-java/pull/501/files#diff-f4a32244a27971a081fa72ffb340ca67R113